pulsar-common/src/main/java/com/yahoo/pulsar/common/util/collections/ConcurrentLongPairSet.java

```diff
@@ -528,4 +529,23 @@ public int compareTo(LongPair o) {
             }
         }
     }
+    
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append('{');
+        final AtomicBoolean first = new AtomicBoolean(true);
```

Comment:
the lambda doesn't run concurrently, though from a lambda you cannot modify a `boolean` flag that's defined outside the lambda (like in c++ you pass references into a lambda.. :) ). So using the `AtomicBoolean` is a trick to circumvent it, basically you need to have an object containing a boolean that you can reference and modify.